File: zmqTest.py
import zmq, time

context = zmq.Context()

#  Socket to talk to server
print("Connecting to hello world server...")
socket = context.socket(zmq.PUSH)
socket.connect("tcp://localhost:5555")

#  Send 10 messages to the server
for request in range(10):
    print(f"Sending request {request} ...")
    socket.send_string(f"Hello{request}\0")

    time.sleep(2)

File: zmqTest.c

#include <zmq.h>
#include <stdio.h>
#include <unistd.h>
#include <string.h>
#include <assert.h>

int main (void)
{
    //  Socket to talk to clients
    void *context = zmq_ctx_new ();
    void *responder = zmq_socket (context, ZMQ_PULL);
    int rc = zmq_bind (responder, "tcp://*:5555");
    assert (rc == 0);

    while (1) {
        char buffer [10];
        zmq_recv (responder, buffer, 10, 0);
        if (strlen(buffer) > 0)
        {
            printf ("Received %s\n", buffer);
        }
        sleep (1);          //  Do some 'work'
        strcpy(buffer, "");
        
    }
    return 0;
}
